warehouse/query-core/src/main/java/datawave/query/util/sortedset/HdfsBackedSortedSet.java

```diff
@@ -162,16 +165,42 @@ public boolean isValid() {
             FileSystem fs = getFs();
             Path uniqueDir = getUniqueDir();
             
-            // attempt to create the folder if it doesn't exist
-            if (!fs.exists(uniqueDir) && !fs.mkdirs(uniqueDir))
-                throw new IOException("Unable to create directory [" + uniqueDir + "] in filesystem [" + fs + "]");
+            // Lazily create the required ivarator cache dirs.
+            ensureDirsCreated();
             
             // generate a unique file name
             fileCount++;
             Path file = new Path(uniqueDir, FILENAME_PREFIX + fileCount + '.' + System.currentTimeMillis());
             return new SortedSetHdfsFileHandler(fs, file, persistOptions);
         }
         
+        private void ensureDirsCreated() throws IOException {
+            IvaratorCacheDirConfig config = ivaratorCacheDir.getConfig();
+            if (config.isValid()) {
+                ensureCreation(new Path(ivaratorCacheDir.getPathURI()));
+                ensureCreation(getUniqueDir());
+            } else {
+                throw new IOException("Unable to create Ivarator Cache Dir for invalid config: " + config);
+            }
+        }
+        
+        private void ensureCreation(Path path) throws IOException {
+            try {
+                FileSystem fs = getFs();
+                if (!fs.exists(path)) {
+                    // Attempt to create the required directory if it does not exist.
+                    if (!fs.mkdirs(path)) {
+                        throw new IOException("Unable to mkdirs: fs.mkdir(" + path + ")->false");
```

Comment:
If we cannot create the directory for whatever reason, the we need to be able to try the next directory in the configuration.

Reply:
@ivakegg could you elaborate more on this? I'm not sure what you mean by the next directory. I see that `IvaratorCacheDirConfig` has a basePathURI. Are you referring to trying to create multiple unique paths?

Reply:
Looking through the code it appears that the logic presumes that we can create the directory on the first valid ivarator cache dir config anyway (there can be multiple set on the QueryIterator).  I take back this comment.  I think we are good.

Reply:
Okay, great!